lib/operations/count.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Aspect = require('./operation').Aspect;
+const buildCountCommand = require('./common_functions').buildCountCommand;
+const defineAspects = require('./operation').defineAspects;
+const OperationBase = require('./operation').OperationBase;
+
+class CountOperation extends OperationBase {
+  constructor(cursor, applySkipLimit, options) {
+    super(options);
+
+    this.cursor = cursor;
+    this.applySkipLimit = applySkipLimit;
+  }
+
+  execute(callback) {
+    const cursor = this.cursor;
+    const applySkipLimit = this.applySkipLimit;
+    const options = this.options;
+
+    if (applySkipLimit) {
+      if (typeof cursor.cursorSkip() === 'number') options.skip = cursor.cursorSkip();
+      if (typeof cursor.cursorLimit() === 'number') options.limit = cursor.cursorLimit();
+    }
+
+    // Ensure we have the right read preference inheritance
+    if (options.readPreference) {
+      cursor.setReadPreference(options.readPreference);
+    }
+
+    if (
+      typeof options.maxTimeMS !== 'number' &&
+      cursor.s.cmd &&
+      typeof cursor.s.cmd.maxTimeMS === 'number'
+    ) {
+      options.maxTimeMS = cursor.s.cmd.maxTimeMS;
+    }
+
+    let finalOptions = {};
+    finalOptions.skip = options.skip;
+    finalOptions.limit = options.limit;
+    finalOptions.hint = options.hint;
+    finalOptions.maxTimeMS = options.maxTimeMS;
+
+    // Command
+    const delimiter = cursor.s.ns.indexOf('.');
+    finalOptions.collectionName = cursor.s.ns.substr(delimiter + 1);
+
+    let command;
+    try {
+      command = buildCountCommand(cursor, cursor.s.cmd.query, finalOptions);
+    } catch (err) {
+      return callback(err);
+    }
+
+    // Set cursor server to the same as the topology
+    cursor.server = cursor.topology.s.coreTopology;
+
+    // Execute the command
+    cursor.s.topology.command(
```

Comment:
subclass `CommandOperation`?

Reply:
This can't currently be a subclass of `CommandOperation` because `CommandOperation` has quite a bit of database-specific operations, like [this](https://github.com/mongodb/node-mongodb-native/commit/97ed9da76b7ed10e0988f84e0fe28f08a6306b9d#diff-d7027a65bc9a9149aba8ddb731de42fcR46). I could separate `CommandOperation` into something like `DbCommandOperation` and `CommandOperation` if that would be preferable. Alternatively, I can make a ticket for this and get to it after all operations are in their own classes.